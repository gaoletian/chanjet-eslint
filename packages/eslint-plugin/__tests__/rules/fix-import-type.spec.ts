import { createESlint } from '../testUtil';

const onlytext = (str: string) => str.replace(/(\s|\n)/g, '');
const fsMock = {
  appendFile: jest.fn(),
  ensureFile() {
    return true;
  },
  readFile() {
    return `export type {Foo} from 'src/module/foo`;
  },
};

const delay = (duration: number) => new Promise((resolve) => setTimeout(resolve, duration));

describe('fix-import-type', () => {
  beforeEach(() => {
    jest.mock('fs-extra', () => {
      return fsMock;
    });
  });
  afterEach(() => {
    jest.unmock('fs-extra');
  });
  test('fix-import-type should work', async () => {
    const code = [
      `import type {Foo} from 'src/module/foo';`,
      `import type Bar from 'src/module/bar';`,
      `import type {default as Baz} from 'src/module/baz';`,
      `let foo: Foo = null;`,
      `let bar: Bar = null;`,
      `let baz: Baz = null;`,
    ].join('');

    const codeFixed = [
      `let foo: StoresExternal.Foo = null;`,
      `let bar: StoresExternal.Bar = null;`,
      `let baz: StoresExternal.Baz = null;`,
    ].join('');

    const linter = createESlint({
      'chanjet/fix-import-type': 'error',
    });

    const filePath = '/root/src/stores/appStore.ts';
    let result = await linter.lintText(code, { filePath });
    expect(onlytext(result[0].output as string)).toBe(onlytext(codeFixed));
    await delay(60);
    expect(fsMock.appendFile).toBeCalled();
    expect(fsMock.appendFile).toMatchSnapshot();
  });
});
