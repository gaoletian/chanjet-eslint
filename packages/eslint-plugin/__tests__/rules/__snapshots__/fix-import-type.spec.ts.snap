// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fix-import-type fix-import-type should work 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/root/src/types/StoresExternal.d.ts",
      "export type {Foo} from 'src/module/foo'
",
    ],
    Array [
      "/root/src/types/StoresExternal.d.ts",
      "export type {default as Bar} from 'src/module/bar'
",
    ],
    Array [
      "/root/src/types/StoresExternal.d.ts",
      "export type {default as Baz} from 'src/module/baz'
",
    ],
    Array [
      "/root/src/types/StoresExternal.d.ts",
      "// this file is auto generate 
export as namespace StoresExternal;
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
